### PR TITLE
A pandas-like table API

### PR DESCRIPTION
**Pull request by xuebinsu**
More details will be added later.

### CODE REVIEW

File: greenplumpython/table.py [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+        """
+        Returns
+        - a Column of the current Table if key is string, or
+        - a new Table from the current Table per the type of key:
+            - if key is a list, then SELECT a subset of columns, a.k.a. targets;
+            - if key is an Expr, then SELECT a subset of rows per the value of the Expr;
+            - if key is a slice, then SELECT a portion of consecutive rows
+        """
+        if isinstance(key, str):
+            return Column(key, self)
```
Comment (beeender) [inline greenplumpython/table.py:58]: So the column will be a special `Table` which contains one column only?

File: greenplumpython/table.py [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+                f"SELECT * FROM {self.name} LIMIT {key.stop} {offset_clause}",
+                parents=[self],
+            )
+
+    def list_lineage(self) -> Iterable["Table"]:
+        lineage = [self]
+        current = 0
+        while current < len(lineage):
+            for table in lineage[current].parents:
+                lineage.append(table)
```
Comment (yihong0618) [inline greenplumpython/table.py:80]: is logic here len(lineage) always==1?

Comment (xuebinsu) [reply greenplumpython/table.py:80]: No. More items will be appended to the list as the loop goes.

Comment (yihong0618) [reply greenplumpython/table.py:80]: cool missed thanks

File: greenplumpython/table.py [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+    ) -> None:
+        self.query = query
+        self.parents = parents
+        if table_name is not None:
+            self.name = table_name
+        else:
+            self.name = "tb_" + "".join(
+                random.choice(string.ascii_lowercase) for _ in range(60)
+            )
+        if any(parents):
```
Comment (yihong0618) [inline greenplumpython/table.py:43]: cool

**Review by yihong0618 — APPROVED** [review]
LGTM

**Pull request closed without merging**